### ISSUE TITLE
Owned entity of detached entity is not repopulated properly after query in same context

### ISSUE THREAD

**Issue by Hoffs** [dotnet/efcore#19659]
After detaching an entity which has an owned entity doing the query for same entity in the same context does not populate owned entity fields. If owned entity is detached as well, then owned field is populated properly. If using `AsNoTracking` field is populated properly as well. The generated query is also correct and includes all fields that are needed to populated owned entity.
I believe any of these methods should populate owned field properly with the actual value.

### Steps to reproduce
Fully contained reproducible program:
``` C#
class Program
    {
        public class BloggingContext : DbContext
        {
            public BloggingContext([NotNull] DbContextOptions options) : base(options) { }
            public DbSet<Blog> Blogs { get; set; }
        }

        public class Blog
        {
            public int Id { get; set; }
            public string Name { get; set; }
            public BlogInfo Info { get; set; }
        }

        [Owned]
        public class BlogInfo
        {
            public string Nested { get; set; }
        }

        static async Task Main(string[] args)
        {
            using var connection = new SqliteConnection("DataSource=:memory:");
            connection.Open();
            using var ctx = new BloggingContext(new DbContextOptionsBuilder<BloggingContext>().UseSqlite(connection).Options);
            ctx.Database.EnsureCreated();
            var ent = new Blog { Id = 1, Name = "Name", Info = new BlogInfo { Nested = "NameInside" } };

            await ctx.AddAsync(ent);
            await ctx.SaveChangesAsync();

            Console.WriteLine($"Before detach: {ctx.Blogs.Local.Count}");
            ctx.Entry(ent).State = EntityState.Detached;
            // ctx.Entry(ent.Info).State = EntityState.Detached; // This makes `FirstAsync` populate `Info` field properly.

            Console.WriteLine($"After detach: {ctx.Blogs.Local.Count}");
            var createdEnt = await ctx.Blogs.FirstAsync();

            Console.WriteLine($"After query: {ctx.Blogs.Local.Count}");
            Console.WriteLine($"Local entity: {JsonSerializer.Serialize(ent)}");
            Console.WriteLine($"Queried entity: {JsonSerializer.Serialize(createdEnt)}");

            var createdEntNotTracking = await ctx.Blogs.AsNoTracking().FirstAsync();
            Console.WriteLine($"Queried AsNoTracking entity: {JsonSerializer.Serialize(createdEntNotTracking)}");
        }
    }
```
```
Output:
Before detach: 1
After detach: 0
After query: 1
Local entity: {"Id":1,"Name":"Name","Info":{"Nested":"NameInside"}}
Queried entity: {"Id":1,"Name":"Name","Info":null}
Queried AsNoTracking entity: {"Id":1,"Name":"Name","Info":{"Nested":"NameInside"}}
```

### Further technical details

EF Core version: 3.1.1
Database provider: Microsoft.EntityFrameworkCore.Sqlite
Target framework:  .NET Core 3.0
Operating system: Windows 10

**Comment by ajcvickers** [comment]
Note for team: looks like a bug in query\identity resolution\owned types\and fixup.

**Comment by jonathanantoine** [comment]
Hello, 

This can't be fixed in a Service Release ?

**Comment by ajcvickers** [comment]
@jonathanantoine We will consider it. However, see the [release planning process](https://docs.microsoft.com/en-us/ef/core/what-is-new/release-planning) for details on how we decide whether or not to patch,

**Comment by ajcvickers** [comment]
Verified that this has the same root cause as #19203 and is therefore already fixed in the upcoming 3.1.3 release.
